README.md

```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
+* **UpgradeCannotStart**
+  * In this state, the upgrade cannot start because one the following reasons:
+    * Blocking CRs are missing from the system
+    * Blocking CRs have not yet reached the **UpgradeCompleted** state
 * **UpgradeNotCompleted**
-  * In this state, the controller will enforce the policies following the remediation plan built in the **UpgradeNotStarted** state
-  * The policies of subsequent batches will start as soon as the current batch policies are all compliant unless the batch times out, in which case the controller will move on to the next batch. The value of this batch timeout is the **ClusterGroupUpgrade** timeout divided by the remediation plan number of batches.
-  * Within a batch, the platform upgrade will be enforced first. Once the platform upgrade policy is compliant, then the operator upgrades policies will be enforced
+  * In this state, the controller will make enforce copied of the inform *managedPolicies* policies and adds clusters to the corresponding placement rules following the remediation plan built in the **UpgradeNotStarted** state.
+  * Enforcing the policies for subsequent batches starts immediately after all the clusters of the current batch are compliant with all the *managedPolicies*. If the current batch times out, then the controller moves on to the next batch. The value for the batch timeout is the **ClusterGroupUpgrade** timeout divided by the number of batches from the remediation plan.
   * The controller will transition to **UpgradeTimedOut** state in two cases:
     * If the **ClusterGroupUpgrade** has the first batch as canaries and the policies for this first batch are not compliant within the batch timeout
     * If the policies for the upgrade have not turned to compliant within the *timeout* value specified in the *remediationStrategy*
 * **UpgradeTimedOut**
   * In this state, the controller will periodically check if all the policies for the **ClusterGroupUpgrade** are compliant, and in that case it will transition to **UpgradeCompleted**. This is to give a chance for upgrades to catch up, as they could be taking long to complete due to network, CPU or other issues but they are not really stuck and they can indeed complete
 * **UpgradeCompleted**
   * In this state, the upgrades of the clusters are complete
-  * If the *deleteObjects* field is set to **true**, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful
+  * If the *action.afterCompletion.deleteObjects* field is set to **true** by default, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful.
+* **PrecachingRequired**
```

Comment:
Move this section up?

Reply:
I'd rather leave it here and rephrase to
> If the *action.afterCompletion.deleteObjects* field is set to **true**, the controller will delete the underlying RHACM objects (policies, placement bindings, placement rules, managed cluster views) once the upgrade completes.

Reply:
Sorry, I actually meant the Precaching state section. It's probably better to have the states in actual order?